spec/text-editor-spec.coffee

```diff
@@ -211,6 +211,50 @@ describe "TextEditor", ->
         editor.getLastCursor().destroy()
         expect(editor.getLastCursor().getBufferPosition()).toEqual([0, 0])
 
+    describe ".getCurrentWordBufferRange()", ->
+      cursor = null
+
+      beforeEach ->
+        editor.setText('atom-+electron')
+        atom.config.set 'editor.nonWordCharacters', '-+'
+        cursor = editor.getLastCursor()
+
+      describe "when cursor is between two word characters", ->
+        beforeEach ->
+          editor.setCursorBufferPosition([0, 2]) # at|om-+electron
+
+        describe "when includeNonWordCharacters is false", ->
+          it "returns the word under the cursor, defined with word characters", ->
+            expect(cursor.getCurrentWordBufferRange({"includeNonWordCharacters": false})).toEqual [[0, 0], [0, 4]] # atom
+
+        describe "when includeNonWordCharacters is true", ->
+          it "returns the word under the cursor, defined with word characters", ->
+            expect(cursor.getCurrentWordBufferRange({"includeNonWordCharacters": true})).toEqual [[0, 0], [0, 4]] # atom
+
+      describe "when cursor is between two non-word characters", ->
+        beforeEach ->
+          editor.setCursorBufferPosition([0, 5]) # atom-|+electron
+
+        describe "when includeNonWordCharacters is false", ->
+          it "returns a zero-width range", ->
+            expect(cursor.getCurrentWordBufferRange({"includeNonWordCharacters": false})).toEqual [[0, 5], [0, 5]] # <nothing>
+
+        describe "when includeNonWordCharacters is true", ->
+          it "returns the word under the cursor, defined with non-word characters", ->
+            expect(cursor.getCurrentWordBufferRange({"includeNonWordCharacters": true})).toEqual [[0, 4], [0, 6]] # -+
+
+      describe "when cursor is between a word character and a non-word character", ->
+        beforeEach ->
+          editor.setCursorBufferPosition([0, 6]) # atom-+|electron
+
+        describe "when includeNonWordCharacters is false", ->
+          it "returns the word next to the cursor, defined with word characters", ->
+            expect(cursor.getCurrentWordBufferRange({"includeNonWordCharacters": false})).toEqual [[0, 6], [0, 14]] # electron
+
+        describe "when includeNonWordCharacters is true", ->
+          it "returns the word next to the cursor, defined with word characters", ->
+            expect(cursor.getCurrentWordBufferRange({"includeNonWordCharacters": true})).toEqual [[0, 6], [0, 14]] # electron
+
```

Comment:
> Do you think we could express the specs in terms of `editor.getWordUnderCursor`? Seems like we should be able to drop comments with that, and to let specs tell a better story. :thought_balloon:

That's what my first version of these specs looked like, and then I decided to switch them to use `cursor.getCurrentWordBufferRange` directly. My plan was actually to add spec for both methods because I believe that every public API method should have it's own specs, even in cases when one public API method is mostly a wrapper for another (as is the case here -- both are public, and one is mostly a wrapper for the other). I think that approach will help us feel safer about not breaking any behavior and breaking our promise to package developers. I can definitely imagine situations where changes are made to both methods and we find out that tests for `editor.getWordUnderCursor` still pass, but `cursor.getCurrentWordBufferRange` is not covered at all anymore or an edge case is introduced which the tests for `editor.getWordUnderCursor` don't cover.  

Still, this issue and pull request have been a big learning experience for me :school_satchel:, and I would love to hear what everyone thinks about my reasoning above. /cc @atom/feedback Thanks for bringing this up, @as-cii. :blue_heart: